pages/index.tsx

```diff
@@ -1,17 +1,36 @@
 import { loadProfiles } from "../util/load-profiles";
 import Head from "next/head";
-import OverviewCard from "../components/overview-card";
 import { ProfileType } from "../types/ProfileType";
+import SearchFilter from "../components/SearchFilter";
+import { useCallback, useEffect, useState } from "react";
+import OverviewCard from "../components/Profile";
 
 export default function Home({ profiles }: { profiles: ProfileType[] }) {
+  const [searchString, setSearchString] = useState("")
+  const [filteredProfiles, setFilteredProfiles] = useState<ProfileType[]>([])
+
+  const filterBySearchString = useCallback(() => {
+    if(searchString !== ""){
+      const filteredProfiles = profiles.filter(profile => profile.name.toLowerCase().includes(searchString.toLowerCase()) || profile.location.toLowerCase().includes(searchString.toLowerCase()))
+      return setFilteredProfiles(filteredProfiles)
```

Comment:
unsure what the return is for since ur just setting state

Reply:
return is void. should I add :void? to declare return type.

Reply:
you could just remove the return part and it will be fine